cryostatio/cryostat-operator
# feat(insights): use separate deployment for Insights

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #961 
Depends on: https://github.com/cryostatio/cryostat-agent/issues/510

## Description of the change:
* Removes Go module dependency on the Runtimes Inventory Operator
* Adds a Kustomize overlay to optionally build a bundle that includes a second deployment for the Runtimes Inventory Operator container image with its own RBAC
* Bumps Kustomize version to support Replacements

## Motivation for the change:
* Removes a Go module dependency that is only useful in downstream Red Hat builds
* When enabled. the Insights integration no longer counts towards Cryostat Operator's resource limits

## How to manually test:
1. `make bundle bundle-build ENABLE_INSIGHTS=true`
2. Deploy an agent sample app with `-Drht.insights.java.debug=true`
3. Check for a line in the agent log like: `Red Hat Insights - Payload was accepted for processing`
